### PR TITLE
Compare technologies

### LINKED ISSUE #166: [outputs] add plotting function that compares pv technologies
a plotting functions that reads loop-outputs from different scenarios and plots them over an equal set of (variable) KPI's.
Analogue to the plot_kpi_loop() function, only that it plots different technologies / with or without sector coupling.

### PR DESCRIPTION
**Pull request by Piranias**
Fix #166

**Changes proposed in this pull request**:
- new function in outputs.py: compare_weather_years() that plots a barplot for several weather year with bars for dni, ghi, ghi and temp
- adapt loop_KPI function for more technologies / indicators. This way several output scenarios that include a loop over the same variable / steps can be compared.
- The function is designed to compare si/psi/cpv technologies, but can be used for other comparisons as well (e.g. location (spain, germany) or with or without hp.

The following steps were realized, as well (if applies):
- [x] Use in-line comments to explain your code
- [x] Write docstrings to your code
- [:x: ] For new functionalities: Explain in readthedocs
- [x] Write test(s) for your new patch of code
- :x: Update the CHANGELOG.md (let's start this after the first release)
- [x] Apply black (`black . --exclude docs/`)
- [x] Check if full simulation tests pass locally (`EXECUTE_TESTS_ON=master pytest`)

*Please mark above checkboxes as following:*
- [ ] Open
- [x] Done

:x: Check not applicable to this PR

<sub>*For more information on how to contribute check the [CONTRIBUTING.md](https://github.com/rl-institut/mvs_eland/blob/dev/CONTRIBUTING.md).*<sub>

### CODE REVIEW

**Comment by Piranias** [comment]
I think one reviewer is enough, whoever has time :)

File: pvcompare/outputs.py [hunk 1]
```diff
@@ -642,15 +642,15 @@ def plot_kpi_loop(
             "self consumption",
             "self sufficiency",
             "Degree of autonomy"
-    scenario_name: str
-        Name of the Scenario. The name should follow the scheme:
-        "Scenario_A1", "Scenario_A2", "Scenario_B1" etc.
+    scenario_dict: dictionary
+        dictionary with the scenario Names that should be compared as keys and
```
Comment (Piranias) [inline pvcompare/outputs.py:646]:
```suggestion
        dictionary with the scenario names that should be compared as keys and
```

File: pvcompare/outputs.py [hunk 2]
```diff
@@ -642,15 +642,15 @@ def plot_kpi_loop(
...
+        label for that scenario as value. e.g.: {"Scenario_A1" : "si", "Scenario_A2": "cpv"}
```
Comment (Piranias) [inline pvcompare/outputs.py:647]:
```suggestion
        a label for the scenario as value. e.g.: {"Scenario_A1" : "si", "Scenario_A2": "cpv"}
```

File: pvcompare/outputs.py [hunk 3]
```diff
@@ -642,15 +642,15 @@ def plot_kpi_loop(
...
+         Notice: allscenarios you want to compare, need to include the
```
Comment (Piranias) [inline pvcompare/outputs.py:648]:
```suggestion
         Notice: all scenarios you want to compare, need to include the
```

File: pvcompare/outputs.py [hunk 4]
```diff
@@ -642,15 +642,15 @@ def plot_kpi_loop(
...
+         loop-outputs for the same variable / steps. The scenario names name
```
Comment (Piranias) [inline pvcompare/outputs.py:649]:
```suggestion
         loop-outputs for the same variable / steps. The scenario names
```

**Review by SabineHaas — COMMENTED** [review]
I haven't fully reviewed, yet. My first comments:
- remove plot in test outputs e.g. via `tear_down_module` -method, -class whichever you need
- When I tried to test the code I realized that in `building_paramters.csv` the file names for electricity consumption end with `.csv` while in `static_inputs` they are `.xlsx` files.

**Comment by Piranias** [comment]
> I haven't fully reviewed, yet. My first comments:
> 
>     * remove plot in test outputs e.g. via `tear_down_module` -method, -class whichever you need
> 
>     * When I tried to test the code I realized that in `building_paramters.csv` the file names for electricity consumption end with `.csv` while in `static_inputs` they are `.xlsx` files.

I guess the error in building_parameters.csv is because "user_inputs" is not being pushed into the dev. It's the folder that each one has to change/work on their own. Can you or someone review the rest? this is closely linked to another PR i'm working on right now (issue #219 )

**Comment by SabineHaas** [comment]
> > I haven't fully reviewed, yet. My first comments:
> > ```
> > * remove plot in test outputs e.g. via `tear_down_module` -method, -class whichever you need
> > 
> > * When I tried to test the code I realized that in `building_paramters.csv` the file names for electricity consumption end with `.csv` while in `static_inputs` they are `.xlsx` files.
> > ```
> 
> I guess the error in building_parameters.csv is because "user_inputs" is not being pushed into the dev. It's the folder that each one has to change/work on their own. 

Yes sorry, I thought it was wrong in user collections, as well, but there it's right.

Can you or someone review the rest? this is closely linked to another PR i'm working on right now (issue #219 )

**Pull request merged**